python/nammu/view/AtfAreaView.py

```diff
@@ -402,8 +403,36 @@ def undoableEditHappened(self, event):
             # to false. Note undo() only undoes compound edits when they
             # are not in progress.
             self.current_compound.end()
-            self.current_compound = CompoundEdit()
+            self.current_compound = AtfCompoundEdit()
             self.undo_manager.addEdit(self.current_compound)
 
         # Always add current edit to current compound
         self.current_compound.addEdit(edit)
+
+
+class AtfCompoundEdit(CompoundEdit):
+    # add a getter for a protected field
+    def getEdits(self):
+        field = CompoundEdit.getDeclaredField("edits")
+        field.setAccessible(True)
+        return field.get(self)
+
+    def firstEdit(self):
+        return self.getEdits()[0]
+
+
+class AtfUndoManager(UndoManager):
+    def __init__(self, panel):
+        self.panel = panel
+
+    def editToBeRedone(self):
+        """
+        Return the protected method `editToBeRedone()`.
```

Comment:
Minor, but:
```suggestion
        Expose the protected method `editToBeRedone()`.
```
(since we're not really returning the method itself)